From 57b7f09788943c088bd15260c31f459ecb813966 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Fri, 9 Jan 2015 14:22:05 +0530
Subject: [PATCH 3/3] power_supply: Change current based on
 USB_EVENT_ENUMERATED

On successful enumeration USB driver sends the event
USB_EVENT_ENUMERATED with the current value. Use this to
configure input current for charger

Change-Id: I0b53026ce7761599b33ae3ff0fb132d29b61f5c2
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/power_supply_charger.c | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 5422826..889fe69 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -162,11 +162,21 @@ static int handle_cable_notification(struct notifier_block *nb,
 {
 	struct power_supply_cable_props cap;
 
-	if (event != USB_EVENT_CHARGER && event != PSY_CABLE_EVENT)
+	if (event != USB_EVENT_CHARGER && event != PSY_CABLE_EVENT &&
+				event != USB_EVENT_ENUMERATED)
 		return NOTIFY_DONE;
 
-	if (data)
-		memcpy(&cap, data, sizeof(struct power_supply_cable_props));
+	if (!data)
+		return NOTIFY_DONE;
+
+	if (event == USB_EVENT_ENUMERATED) {
+		cap.chrg_type = POWER_SUPPLY_CHARGER_TYPE_USB_SDP;
+		cap.chrg_evt = POWER_SUPPLY_CHARGER_EVENT_CONNECT;
+		cap.ma = *(int *)data;
+	} else {
+		cap = *(struct power_supply_cable_props *)data;
+	}
+
 	process_cable_props(&cap);
 
 	return NOTIFY_OK;
-- 
1.9.1

